src/walletConnect/v2/reducer.ts

```diff
@@ -80,7 +80,20 @@ export const reducer = (
             pendingAction.id !== action.request.id && pendingAction.topic !== action.request.topic
         ),
       }
-
+    case Actions.REMOVE_EXPIRED_SESSIONS_V2: {
+      const unexpiredSessions = state.sessions.filter(
+        (session) => session.expiry > action.dateInSeconds
+      )
+      return {
+        ...state,
+        sessions: unexpiredSessions,
```

Comment:
Shouldn't we also check `pendingSessions`?

Reply:
good catch! i didn't notice the expiry param before, i added now!